sherpa/utils/__init__.py

```diff
@@ -2736,6 +2737,56 @@ def parallel_map(function, sequence, numcores=None):
 
     return run_tasks(procs, err_q, out_q, numcores)
 
+def parallel_map_funcs(funcs, datasets, numcores=None):
+
+    if not numpy.iterable(funcs):
+        raise TypeError("input '%s' is not iterable" % repr(funcs))
+
+    if not numpy.iterable(datasets):
+        raise TypeError("input '%s' is not iterable" % repr(datasets))
+
+    for func in funcs:
+        if not callable(func):
+            raise TypeError("input func '%s' is not callable" % repr(func))
+
+    funcs_size = len(funcs)
+    datasets_size = len(datasets)
+    if funcs_size != datasets_size:
+        msg = "input funcs (%d) and datsets (%d) size must be same" % \
+            (funcs_size, datasets_size)
+        raise TypeError(msg)
+
+    if not _multi or datasets_size == 1 or \
+            (numcores is not None and numcores < 2):
```

Comment:
Is there a usecase for negative numcores or are you just catching both 0 and 1 here?

Reply:
users are creative with their options

Reply:
So, how about `numcores == 1`? (I know you copied that just from the function above, but I still wonder why it's written this way.

Reply:
I experimented with @hamogu's suggestion:

```
@@ -2757,7 +2835,7 @@ def parallel_map_funcs(funcs, datasets, numcores=None):
         raise TypeError(msg)
 
     if not _multi or datasets_size == 1 or \
-            (numcores is not None and numcores < 2):
+            (numcores is not None and numcores == 1):
         return list(map(funcs[0], datasets))
```
I'm using the script and data that @anetasie gave me, the one that makes life more miserable for the user if using multicores rather than a single core because of the large overhead in multi-processing:

```
(sherpa) [dtn@devel12 sherpa]$ python fit_3_spectra.py.orig 1
...
fit with numcores == 1 took 24.588085412979126 secs

(sherpa) [dtn@devel12 sherpa]$ python fit_3_spectra.py.orig 3
...
fit with numcores == 3 took 53.463428020477295 secs
```
If the code is left unchanged: only one core is used; However, if the change as suggested by @hamogu the code runs with multiple cores (don't ask me how many cores were used):

```
(sherpa) [dtn@devel12 sherpa]$ python fit_3_spectra.py.orig -1
...
fit with numcores == -1 took 54.06742238998413 secs

(sherpa) [dtn@devel12 sherpa]$ python fit_3_spectra.py.orig 0
...
fit with numcores == 0 took 53.430519342422485 secs
```

So if ```numcores == 1``` is to be substituted for ```numcores < 2```, perhaps one should also add in the following check:

```
if numcores <= 0:
    raise ValueError('numcores must be > 0')
```

Reply:
I guess 0 has some special meaning (maybe "default"). Anyway, it was just a suggestion. IF you have good reason to do it this way, I'm fine with it, too.